src/GNDStk/Tree/src/init.hpp

```diff
@@ -0,0 +1,93 @@
+
+// -----------------------------------------------------------------------------
+// Tree::init()
+// For making or modifying trees
+// -----------------------------------------------------------------------------
+
+// init(top-level node, format[, version[, encoding]])
+// Example:
+//    GNDStk::Tree<> t(GNDStk::child::reactionSuite,GNDStk::format::xml);
+// Briefly:
+//    Tree<> t(reactionSuite,format::xml);
+// with appropriate usings.
+
+
+// ------------------------
+// init(top, format, ...)
+// ------------------------
+
+template<class RESULT, bool MULTIPLE, class METADATA, class CHILDREN>
+nodeType &init(
```

Comment:
Why an `init` function instead of just a default constructor?

Reply:
I do have a default constructor, which at the moment creates an absolutely empty tree structure. Then I have a couple of what I call "starter tree" constructors, which actually just forward to init(). An init(), individually, is useful if somebody has a tree structure already and, for whatever reason, wants to (re)initialize it with the basics of a to-be-built GNDS hierarchy. It's probably most likely that someone would just create a new tree in order to do this, but I could also see someone wanting to restart an existing one for whatever reasons. (I've seen lots of different style preferences from users.) The init()s contain contents that the two "starter tree" constructors would otherwise need to contain, in lieu of their current one-liners, so my reasoning was that having the init()s adds a small convenience for people who want it, with minimal extra complexity.

Reply:
So when I crate a `Tree` object, I then have to call the `init()` function? That makes it more awkward to use and I'm bound to forgot to call `init()` at some point and wonder why things aren't working.

Reply:
No worries, I wouldn't make anyone do anything like that. :-) You create Tree objects in the customary and expected way - by calling a constructor. init() is a way to sort of re-initialize a tree with basic contents, like the top-level GNDS node. Think of it as being a bit like std::vector's clear() function. Clear out a vector, because, for whatever reason, you'd like to use the same vector (not construct a new one), but remove all the existing contents. Tree has a clear(), too; the difference is that init() clears, but then also adds basic stuff that you'll need at the top level of the tree structure.

Reply:
Along with some other recent changes (not quite ready to be uploaded yet), I renamed this to reset() in order to clarify its meaning.